symphony-bdk-legacy/sms-sdk-renderer-java/src/test/java/it/services/SmsRendererTest.java

```diff
@@ -0,0 +1,136 @@
+    final String jsonMessageContext = this.getStringFromFile("/SmsRenderer/JSONData/"+fileName);
+
+    final String compiledTemplate = this.minifyHTML(SmsRenderer.renderInBot(jsonMessageContext, smsTypes));
+    assertNotNull(compiledTemplate);
+  }
+
+  private void testJsonObject(final SmsRenderer.SmsTypes smsTypes) throws IOException {
+    assertNotNull(smsTypes);
+
+    final String fileName = smsTypes.getName().toLowerCase() + "Json.json";
+    final String jsonMessageContext = this.getStringFromFile("/SmsRenderer/JSONData/"+fileName);
+
+    final JSONObject jsonObject = this.getJsonObject(jsonMessageContext);
+    assertNotNull(jsonObject);
+
+    final String compiledTemplate = this.minifyHTML(SmsRenderer.renderInBot(jsonObject, smsTypes));
+    assertNotNull(compiledTemplate);
+  }
+
+  private String getStringFromFile(final String fileName) throws IOException {
+    assertNotNull(fileName);
+
+    final String jsonMessageContext = this.removeCarriageReturn(this.readResourceContent(fileName));
+    assertNotNull(jsonMessageContext);
+    return jsonMessageContext;
+  }
+
+  private void testJsonArray(final SmsRenderer.SmsTypes smsTypes) throws IOException {
+    assertNotNull(smsTypes);
+
+    final String fileName = smsTypes.getName().toLowerCase() + "JsonArray.json";
+    final String jsonMessageContext = this.getStringFromFile("/SmsRenderer/JSONData/"+fileName);
+
+    final JSONArray jsonArray = new JSONArray();
+    jsonArray.add(jsonMessageContext);
+
+    final String compiledTemplate = this.minifyHTML(SmsRenderer.renderInBot(jsonArray, smsTypes));
+    assertNotNull(compiledTemplate);
+  }
+
+  private String readResourceContent(final String path) throws IOException {
+    assertNotNull(path);
+
+    final InputStream resourceStream = this.getClass().getResourceAsStream(path);
+    assertNotNull(resourceStream);
+
+    final InputStreamReader inputStreamReader = new InputStreamReader(resourceStream);
+    assertNotNull(inputStreamReader);
+
+    final BufferedReader reader = new BufferedReader(inputStreamReader);
+    assertNotNull(reader);
+
+    final StringBuffer sb = new StringBuffer();
+
+    String content;
+    while((content = reader.readLine()) != null){
+      sb.append(content);
+    }
+    return sb.toString();
+  }
+
+  private String minifyHTML(final String text) {
+    return (text==null)?null : this.removeCarriageReturn(text.replaceAll("\\s{2,}", "").replaceAll("\n", "").replaceAll("\r", ""));
+  }
+
+  private String removeCarriageReturn(final String text) {
+    return (text==null)?null : text.replaceAll("\n", "").replaceAll("\r", "");
+  }
+
+  private JSONObject getJsonObject(final String jsonMessageContext) {
+    assertNotNull(jsonMessageContext);
+
+    final JSONParser parser = new JSONParser();
+    JSONObject jsonObject = null;
+    try {
+      jsonObject = (JSONObject) parser.parse(jsonMessageContext);
+    } catch (final ParseException e) {
+      fail(e.getMessage());
```

Comment:
You can remove the try/catch and add a `throws ParseException` in the signature (or a `@SneakyThrows`).